crates/storage/trie/db/libmdbx.rs

```diff
@@ -1,70 +1,90 @@
+use std::marker::PhantomData;
+
 use crate::error::StoreError;
 use libmdbx::{
-    orm::{table, Database},
+    orm::{table, Database, Table},
     table_info,
 };
 
 /// Libmdbx implementation for the TrieDB trait, with get and put operations.
-pub struct Libmdbx(Database);
+pub struct LibmdbxTrieDB<'a, T: Table> {
+    db: &'a Database,
+    phantom: PhantomData<T>,
+}
 
 use super::TrieDB;
 
-table!(
-    /// NodeHash to Node table
-    ( Nodes ) Vec<u8> => Vec<u8>
-);
-
-impl Libmdbx {
-    /// Opens a DB created by a previous execution or creates a new one if it doesn't exist
-
-    pub fn init(trie_dir: &str) -> Result<Self, StoreError> {
-        Self::open(trie_dir).or_else(|_| Self::create(trie_dir))
-    }
-
-    /// Creates a new clean DB
-    pub fn create(trie_dir: &str) -> Result<Self, StoreError> {
-        let tables = [table_info!(Nodes)].into_iter().collect();
-        let path = Some(trie_dir.into());
-        Ok(Self(
-            Database::create(path, &tables).map_err(StoreError::LibmdbxError)?,
-        ))
-    }
-
-    /// Opens a DB created by a previous execution
-    pub fn open(trie_dir: &str) -> Result<Self, StoreError> {
-        // Open DB
-        let tables = [table_info!(Nodes)].into_iter().collect();
-        let db = Database::open(trie_dir, &tables).map_err(StoreError::LibmdbxError)?;
-        Ok(Self(db))
-    }
-
-    #[cfg(test)]
-    /// Creates a temporary DB, for testing purposes only
-    pub fn init_temp() -> Self {
-        use tempdir::TempDir;
-        let tables = [table_info!(Nodes)].into_iter().collect();
-        Self(Database::create(None, &tables).expect("Failed to create temp DB"))
+impl<'a, T: Table> LibmdbxTrieDB<'a, T> {
+    pub fn new(db: &'a Database) -> Self {
```

Comment:
have you tried using an Rc instead of lifetimes?

Reply:
Adding to this: In my PR I had to do some refactors to make the trait work with our StoreEngine api, and I replaced the lifetime with an `Arc`. #399

Reply:
Using an `Rc` would force us to use an `Rc` in the libmdbx `Store`, making it not `Send`

Reply:
@fmoletta so we should merge this one as-is since you make the changes in your PR?

Reply:
Sure